resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -38,5 +41,30 @@ object DbReader {
         .query[Runtime]
         .stream
         .transact(xa)
+
+    override def getBucketsToDelete: Stream[F, BucketToRemove] =
+      sql"""select googleProject, stagingBucket from CLUSTER WHERE status="Deleted" and destroyedDate < now() - interval 15 DAY;"""
+        .query[BucketToRemove]
+        .stream
+        .transact(xa)
+
+    // Return all clusters that has no non "deleted" nodepool
+    override def getK8sClustersToDelete: Stream[F, KubernetesClusterId] =
+      sql"""SELECT kc.id, kc.location, kc.clusterName
+            FROM KUBERNETES_CLUSTER kc 
+            WHERE NOT EXISTS (
+              SELECT *
```

Comment:
I wonder if `SELECT *` causes a lot of data to be read out of the DB or if the DB engine is smart enough to optimize it out. If the former, we may want to specify a column instead of `*` since we don't care about the content.

Reply:
It looks like it's [the latter](https://dev.mysql.com/doc/refman/8.0/en/exists-and-not-exists-subqueries.html)

>Traditionally, an EXISTS subquery starts with SELECT *, but it could begin with SELECT 5 or SELECT column1 or anything at all. MySQL ignores the SELECT list in such a subquery, so it makes no difference.

so you can ignore my previous comment 😊

Reply:
I'm not a SQL expert..this is good to know :)